saleor/dashboard/customer/forms.py

```diff
@@ -4,6 +4,35 @@
 from ...account.models import CustomerNote, User
 
 
+class CustomerDeleteForm(forms.Form):
+
+    def __init__(self, *args, **kwargs):
+        self.instance = kwargs.pop('instance')
+        self.user = kwargs.pop('user')
+        super().__init__(*args, **kwargs)
+
+    def clean(self):
+        data = super().clean()
+        if not self.instance.is_staff:
+            return data
+
+        if self.instance == self.user:
+            raise forms.ValidationError(pgettext_lazy(
+                'Edit customer details in order form error',
+                'You can\'t delete your own account via dashboard, '
+                'please try from the storefront.'))
+        if self.instance.is_superuser:
+            raise forms.ValidationError(pgettext_lazy(
+                'Edit customer details in order form error',
+                'Only superuser can delete his own account.'))
+        can_edit_staff_users = self.user.has_perm('account.edit_staff')
+        if not can_edit_staff_users:
+            raise forms.ValidationError(pgettext_lazy(
```

Comment:
I feel PermissionDenied suits better here
https://docs.djangoproject.com/en/2.0/ref/exceptions/#permissiondenied
Error message could be more generic like `You have no permission to perform this action`

Reply:
I think ValidationError is a better fit for form-related data validation errors

Regarding error messages, I'm not sure what's our stand but prefer like descriptive ones are better for the UX.